src/ui/content/register.tsx

```diff
@@ -1,48 +1,101 @@
-import React from 'react'
+import React, { useState } from 'react'
+import gql from 'graphql-tag'
+import { useMutation } from '@apollo/react-hooks'
 
-type MyProps = { }
-type MyState = { email: string, name: string, password: string, password2: string}
+interface UserCreateInput {
+  name: string,
+  email: string,
+  password: string,
+  age: number,
+}
 
-class RegForm extends React.Component<MyProps, MyState> {
-  constructor(props: any) {
-    super(props);
-    this.state = {email: '', name:'', password:'', password2:''};
+const REG_M = gql`
+  mutation Register($data: UserCreateInput!) {
+    signupUser(data: $data) {
+      name
+    }
+  }
+`
+function RegForm() {
+  const [email, setEmail] = useState('')
+  const [name, setName] = useState('')
+  const [password, setPassword] = useState('')
+  const [password2, setPassword2] = useState('')
 
-    this.handleChange = this.handleChange.bind(this);
-    this.handleSubmit = this.handleSubmit.bind(this);
+  //why cannot I use data:response here? is it a new "data"?
+  const registerCompleted = (data: any) => {
```

Comment:
you can use here any name you like, arguments are names are scoped to the function, they don't exist outside the function that declares them.

You can use any name for the argument, the name goes where data is, after ":" is the type
 ```suggestion
 interface RegisterResponse {
   signupUser {
     name
   }
 }
 
  const registerCompleted = (response: RegisterResponse ) => {
```

Reply:
@lwvb-zoosh Do you mean this?
```
interface RegisterResponse {
   name: string,
}
```

Reply:
```
interface signupResponse {
  name: string,
}

interface RegisterResponse {
  signupUser: signupResponse,
}
```

Is there a better way doing it than this?